src/pages/ProductList/ProductList.js

```diff
@@ -1,57 +1,110 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect, useState } from 'react';
 import './ProductList.scss';
 import ProductCard from './Product/ProductCard';
+import { useParams } from 'react-router-dom';
+import SubCategoryList from './Product/SubCategoryList';
 
 function ProductList() {
+  const [product, setProduct] = useState([]);
+  const [subCategory, setSubCategory] = useState([]);
+
+  const allArr = ['all1'];
+  const medicineArr = [
+    'category1',
+    'subCategory1',
+    'subCategory2',
+    'subCategory3',
+    'subCategory4',
+    'subCategory5',
+  ];
+  const shaverArr = ['category2'];
+
+  const params = useParams();
+  const paramv = params.value;
+  const parami = params.id;
```

Comment:
바로 분해할당 해주면 더 좋을 것 같습니다!
```suggestion
  const { value, id } = useParams();
```